### PR TITLE
Poke at compilation speed a little

### PR DESCRIPTION
**Pull request by axch**

### CODE REVIEW

File: src/lib/Builder.hs [hunk 1]
```diff
@@ -152,6 +153,11 @@ emitAtomToName _ (Var v) = return v
 emitAtomToName hint x = emitHinted hint (Atom x)
 {-# INLINE emitAtomToName #-}
 
+emitAtomToNameAnn :: (Builder r m, Emits n) => NameHint -> LetAnn -> Atom r n -> m n (AtomName r n)
+emitAtomToNameAnn _ _ (Var v) = return v
```
Comment (dougalm) [inline src/lib/Builder.hs:157]: I think it should be valid to put this special case in `EmitDecl` directly (perhaps with the additional check that `LetAnn` is `PlainLet`. That might catch even more redundant renames. Did you try that?

**Review by dougalm — APPROVED** [review]
Looks good, except see comment about adding the `Atom (Var v)` special case to `emitDecl` directly, which might be even better if it works.

File: lib/prelude.dx [hunk 2]
```diff
@@ -1866,6 +1866,7 @@ Not to be confused with a partially applied function
 
 '### Error throwing
 
+@noinline
```
Comment (apaszke) [inline lib/prelude.dx:1869]: It would be sufficient to mark `print`, wouldn't it? `error` is not much larger than that.

File: src/lib/Lower.hs [hunk 3]
```diff
@@ -323,6 +324,7 @@ vectorizeLoops vectorByteWidth abs = liftEnvReaderM do
       Failure errs -> error $ pprint errs
       Success (block', errs) -> return $ (Abs d block', errs)
 {-# INLINE vectorizeLoops #-}
+{-# SCC vectorizeLoops #-}
```
Comment (apaszke) [inline src/lib/Lower.hs:327]: I think you have to take out `INLINE` if you want the `SCC` to work. But it's a good idea to not inline it anyway, so maybe let's do it.

**Review by apaszke — APPROVED** [review]

File: src/lib/Lower.hs (hunk 3 above)
Comment (axch) [reply src/lib/Lower.hs:326]: Done.

File: lib/prelude.dx (hunk 2 above)
Comment (axch) [reply lib/prelude.dx:1870]: Done.

File: src/lib/Builder.hs (hunk 1 above)
Comment (axch) [reply src/lib/Builder.hs]: Good idea.  Done.

**Pull request merged**